### DISCUSSION TITLE
What is the best way to call Python APIs from Next.js?

### DISCUSSION THREAD

**Discussion by cfraboulet** [vercel/next.js#43724] [Help]
Hello 😃 

In order to develop our new project, my team and I are thinking of using Next.js for the frontend with a Python FastAPI backend. While reading the Next.js documentation, I didn't find documentation explaining the different ways to communicate with an (external) Python backend. 

I could see that a simple `fetch` in a `getServerSideProps` function can call any route. But I also saw that we can use the Next.js API routes as a proxy for our FastAPI backend, thus hiding it completely from the frontend in case we want to perform client-side data fetching.

So I wonder what is the best way to communicate with the backend. Using a Next.js API routes as a proxy or not? In which case use it? Only in the client-side data fetching case ?

Thanks for your help 🙏🏻

**Comment by kennedy** [comment]
I also have been thinking of about using fastapi with nextjs

I shall follow this discussion
